src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int amount = 0;
+            for (String field : data) {
+                if ((getDate(field).isAfter(getDate(dateFrom))
+                        || getDate(field).isEqual(getDate(dateFrom)))
+                        && (getDate(field).isBefore(getDate(dateTo))
+                        || getDate(field).isEqual(getDate(dateTo)))
+                        && field.contains(name)) {
+                    amount += getEarnedMoney(field);
+                }
+            }
+            stringBuilder.append(System.lineSeparator()).append(name).append(" - ").append(amount);
+        }
+        return stringBuilder.toString();
+    }
+
+    public LocalDate getDate(String inputData) {
+        String[] items = inputData.split(" ");
```

Comment:
can we split array before `if` block in the method getEarnedMoney()?

Reply:
No.
If we bo that, we must sent dateFrom and dateTo as String array.